### DISCUSSION TITLE
Metal build working in ipad?

### DISCUSSION THREAD

**Discussion by FNsi** [ggml-org/llama.cpp#1763] [Q&A] [answered]
Is it possible to build or offer a wasm file support for iOS?

**Comment by guinmoon** [comment, marked as answer]
After a [slight change in ggml-metal.m](https://github.com/guinmoon/LLMFarm/blob/main/llmfarm_core.swift/Sources/llmfarm_core/llama/ggml-metal.m), I was able to run [this ORCA model](https://huggingface.co/TheBloke/orca_mini_3B-GGML/tree/main) on an iPhone. Works very well.

**Reply by guinmoon** [reply]
With metal, the output is slightly slow at the beginning, but then it works faster than on the CPU. And the phone heats up noticeably less.
